client/src/partials/development/reports/core.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * @description 
+ * Core report controller responsible for downloading report archives and
+ * allowing the user to generate new reports 
+ *
+ * @todo 
+ * Reports should be dynamically templated into the application routes, 
+ * at this point there will be need to link the database report with the client
+ * route
+ *
+ * @todo 
+ * Replace report table with grid after grid analysis is performed (ui grid)
+ *
+ * @todo 
+ * Decide on flexibility of URL structure. Current thoughts are 
+ * report/{key}
+ * report/{key}/{id}
+ * report/{key}/generate
+ * report/{key}/generate/{option_key}/{option_value}
+ */
+angular.module('bhima.controllers').controller('ReportCore', ReportCore);
+
+ReportCore.$inject = ['$routeParams', '$modal', 'ModuleState', 'ReportService'];
+
+function ReportCore($routeParams, $modal, ModuleState, ReportService) { 
+  
+  // Anything assigned to the controller object will be exposed to the template (view)
+  var viewModel = this;
+
+  var state = new ModuleState();
```

Comment:
We've not taken a stand on this yet, but lots of people dislike the `new` keyword in JavaScript.  You can avoid it with by using Factories over Services, which I've been looking into but haven't really changed any of my coding habits yet.  I'd be interested to hear your thoughts in it.

The reason I bring it up with using `Object.defineProperty()` (like [here](https://github.com/IMA-WorldHealth/bhima/pull/830/files#diff-562b483ba8ed35ff924d428aa2bea7cfR34)) is usually the preferred syntax when you are _avoiding_ the `new` constructor.